pkg/assembler/backends/ent/backend/vulnMetadata.go

```diff
@@ -0,0 +1,213 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend
+
+import (
+	"context"
+	stdsql "database/sql"
+	"fmt"
+	"strconv"
+
+	"entgo.io/ent/dialect/sql"
+	"github.com/guacsec/guac/internal/testing/ptrfrom"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent/predicate"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent/vulnerabilityid"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent/vulnerabilitymetadata"
+	"github.com/guacsec/guac/pkg/assembler/backends/ent/vulnerabilitytype"
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+	"github.com/pkg/errors"
+	"github.com/vektah/gqlparser/v2/gqlerror"
+)
+
+func (b *EntBackend) VulnerabilityMetadata(ctx context.Context, filter *model.VulnerabilityMetadataSpec) ([]*model.VulnerabilityMetadata, error) {
+
+	records, err := b.client.VulnerabilityMetadata.Query().
+		Where(vulnerabilityMetadataPredicate(filter)).
+		Limit(MaxPageSize).
+		WithVulnerabilityID(func(q *ent.VulnerabilityIDQuery) {
+			q.WithType()
+		}).
+		All(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve VulnerabilityMetadata :: %s", err)
+	}
+
+	return collect(records, toModelVulnerabilityMetadata), nil
+}
+
+func (b *EntBackend) IngestVulnerabilityMetadata(ctx context.Context, vulnerability model.VulnerabilityInputSpec, vulnerabilityMetadata model.VulnerabilityMetadataInputSpec) (string, error) {
+	recordID, err := WithinTX(ctx, b.client, func(ctx context.Context) (*int, error) {
+		return upsertVulnerabilityMetadata(ctx, ent.TxFromContext(ctx), vulnerability, vulnerabilityMetadata)
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to execute IngestVulnerabilityMetadata :: %s", err)
+	}
+
+	return strconv.Itoa(*recordID), nil
+}
+
+func (b *EntBackend) IngestBulkVulnerabilityMetadata(ctx context.Context, vulnerabilities []*model.VulnerabilityInputSpec, vulnerabilityMetadataList []*model.VulnerabilityMetadataInputSpec) ([]string, error) {
+	var results []string
+	for i := range vulnerabilityMetadataList {
+		hm, err := b.IngestVulnerabilityMetadata(ctx, *vulnerabilities[i], *vulnerabilityMetadataList[i])
+		if err != nil {
+			return nil, gqlerror.Errorf("IngestBulkVulnerabilityMetadata failed with element #%v with err: %v", i, err)
+		}
+		results = append(results, hm)
+	}
+	return results, nil
+}
+
+func vulnerabilityMetadataPredicate(filter *model.VulnerabilityMetadataSpec) predicate.VulnerabilityMetadata {
+	predicates := []predicate.VulnerabilityMetadata{
+		optionalPredicate(filter.ID, IDEQ),
+		optionalPredicate(filter.Timestamp, vulnerabilitymetadata.TimestampGTE),
```

Comment:
there is a disconnect between the backends in terms of how the `timestamp` is stored for verbs (UTC or not). I made an issue to address it: https://github.com/guacsec/guac/issues/1417